asmcli/lib/util.sh

```diff
@@ -42,6 +42,31 @@ run_command() {
   return $RETVAL
 }
 
+######
+# check_curl calls curl and returns non-zero if the http code is not 200 or the
+# curl command fails.
+######
+check_curl() {
+  local TMPFILE; TMPFILE=$(mktemp)
+
+  local HTTPCODE;
+  local RETVAL;
+  HTTPCODE=$(run_command curl --write-out '%{http_code}' --silent --show-error --output "$TMPFILE" "${@}")
+  RETVAL="$?"
+  if [[ "$RETVAL" != "0" ]] ; then
+    rm "$TMPFILE"
+    return "$RETVAL"
+  fi
+  if [[ "$HTTPCODE" != "200" ]] ; then
+    echo "HTTP response code: ${HTTPCODE}"
```

Comment:
Should I use `warn` here, and send the output of $TMPFILE to `warn` as well somehow?

Reply:
For the response code yes. Although now that I'm thinking about it, if it's not 200 do we want to continue the installation? Maybe we should just `fatal` here and call the function `curl_or_fatal` or something?

Reply:
> For the response code yes.
Fixed.

> Although now that I'm thinking about it, if it's not 200 do we want to continue the installation? Maybe we should just fatal here and call the function curl_or_fatal or something?

If all the runIstiod calls fail (since it's called by retry), we will fail the installation. Re: changing this to `curl_or_fatal`, that would make this function harder to use with `retry`. The caller will have enough info to decide to exit the script if that's desired.